pkgs/top-level/haskell-packages.nix

```diff
@@ -130,11 +138,29 @@ in rec {
       ghc = bh.compiler.ghc822;
       compilerConfig = callPackage ../development/haskell-modules/configuration-ghc-8.2.x.nix { };
     };
+    # A variant of GHC 8.2 with https://phabricator.haskell.org/D4388
+    # backported for deterministic profiling symbol names.
+    ghc822DeterministicProfiling = callPackage ../development/haskell-modules {
+      buildHaskellPackages = bh.packages.ghc822DeterministicProfiling;
+      ghc = bh.compiler.ghc822DeterministicProfiling;
+      compilerConfig = callPackage ../development/haskell-modules/configuration-deterministic-profiling.nix {
+        baseConfig = callPackage ../development/haskell-modules/configuration-ghc-8.2.x.nix {};
+      };
+    };
     ghc841 = callPackage ../development/haskell-modules {
       buildHaskellPackages = bh.packages.ghc841;
       ghc = bh.compiler.ghc841;
       compilerConfig = callPackage ../development/haskell-modules/configuration-ghc-8.4.x.nix { };
     };
+    # A variant of GHC 8.4 with https://phabricator.haskell.org/D4388
+    # backported for deterministic profiling symbol names.
+    ghc841DeterministicProfiling = callPackage ../development/haskell-modules {
```

Comment:
Maybe those extra package sets would be better off living in an overlay? Adding them to Nixpkgs proper feels a bit excessive, especially considering the fact we don't even enable profiling builds anywhere by default.

Reply:
IMO they're of general interest (at least when I left last fall Takt was building full profiling sets with distributed builds in hydra, Target is building full profiling sets now, and I'd expect other industrial groups have the same need), but if you insist it's not too big deal for us to carry the set in an overlay.

Reply:
Users who want to build profiling libraries must configure an override to achieve that, so I feel like it's not too much effort for them to override `deterministicProfiling` flag to GHC, too. I'd prefer that use case to be documented in the manual or provided ready-to-use in an overlay, but I'd rather not include it in the main distribution until it's clear that many users actually use it.

Reply:
OK, I'll update this later today.

Reply:
Actually, I'm not sure off the top of my head what the right way is to override a flag to GHC itself in a haskell package set override...